src/typecheck.cpp

```diff
@@ -230,8 +230,8 @@ void typecheck(TypedAST::RecordAccessExpression* ast, TypeChecker& tc) {
 
 	TypeFunctionId dummy_tf = tc.m_core.new_type_function
 	    ( TypeFunctionTag::Record
-	    , {ast->m_member->m_text.str()}
-	    , {{ast->m_member->m_text.str(), member_type}}
+	    , {} // we don't care for fields in dummies
+	    , {{InternedString(ast->m_member->m_text.str()), member_type}}
```

Comment:
`Token::m_text` is actually an InternedString, there is no need to call `.str()` and then make a new `InternedString`. `ast->m_member->m_text` suffices

NIT: "we don't care *about* fields in dummies"

"not care for something" means to dislike that thing

Reply:
NIT: "we don't care *about* fields in dummies"

"not care for something" means to dislike that thing